### PR TITLE
docs: update ApisixRoute docs

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendu@apache.org>

Updates the documentation of ApisixRoute concept.

Once this PR is approved, I will apply similar changes to rest of the concept docs.

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
Any idea why there is a broken link?

File: docs/en/latest/concepts/apisix_route.md [hunk 1]
```diff
@@ -182,15 +174,11 @@ spec:
...
 ```
 
-The above `ApisixRoute` has one route rule, which contains two backends `foo` and `bar`, the weight ratio is `100:50`,
-which means `2/3` requests (with `GET` method and `User-Agent` matching regex pattern `.*Chrome.*`) will be sent to service `foo` and `1/3` requests
-will be proxied to service `bar`.
+## Plugins
 
-Plugins
--------
+APISIX's [70+ Plugins](https://apisix.apache.org/docs/apisix/plugins/batch-requests/) can be used with APISIX Ingress. These Plugins have the same name as in the APISIX documentation.
```
Comment (tao12345666333) [inline docs/en/latest/concepts/apisix_route.md:179]:
```suggestion
APISIX's [80+ Plugins](https://apisix.apache.org/docs/apisix/plugins/batch-requests/) can be used with APISIX Ingress. These Plugins have the same name as in the APISIX documentation.
```

File: docs/en/latest/concepts/apisix_route.md [hunk 2]
```diff
@@ -21,18 +26,13 @@ title: ApisixRoute
...
 
-`ApisixRoute` is a CRD resource which focus on how to route traffic to
-expected backend, it exposes many features supported by Apache APISIX.
-Compared to [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/),
-functions are implemented in a more native way, with stronger semantics.
+`ApisixRoute` is a Kubernetes CRD object that provides a spec to route traffic to services with APISIX. It is much more capable and easy to use compared to the default [Kubernetes Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) resource.
 
-Path based route rules
-----------------------
+See [reference](https://apisix.apache.org/docs/ingress-controller/references/apisix_route_v2/) for the full API documentation.
```
Comment (tao12345666333) [inline docs/en/latest/concepts/apisix_route.md:31]: I don't know why CI judged this link to be unavailable. do you know?  @juzhiyuan @bzp2010

Comment (navendu-pottekkat) [reply docs/en/latest/concepts/apisix_route.md:31]: This is the error. See: https://github.com/apache/apisix-ingress-controller/actions/runs/3270950775/jobs/5381284169#step:3:117

```
ERROR: 1 dead links found!
[✖] file:///github/workspace/docs/en/latest/references/apisix_route_v2/.md → Status: 400
```

File: docs/en/latest/concepts/apisix_route.md (hunk 1 above)
Comment (navendu-pottekkat) [reply docs/en/latest/concepts/apisix_route.md]: Updated.

File: docs/en/latest/concepts/apisix_route.md (hunk 2 above)
Comment (bzp2010) [reply docs/en/latest/concepts/apisix_route.md:31]: Hi, @navendu-pottekkat. Maybe we need to remove the `/` at the end of the URL?

Comment (navendu-pottekkat) [reply docs/en/latest/concepts/apisix_route.md]: It makes sense.

**Review by tao12345666333 — APPROVED** [review]

**Review by lingsamuel — APPROVED** [review]

**Pull request merged**